lib/number.js

```diff
@@ -2,13 +2,37 @@ const { createInterface } = require("readline");
 const fs = require("fs");
 const yaml = require("js-yaml");
 
+function compareSectionNumbers(a, b) {
+  // Split section numbers into parts
+  const partsA = a.split(" ")[0].split(".");
+  const partsB = b.split(" ")[0].split(".");
+
+  // Compare each part of the section numbers
+  for (let i = 0; i < Math.max(partsA.length, partsB.length); i++) {
+    const partA = parseInt(partsA[i]);
+    const partB = parseInt(partsB[i]);
+
+    //TODO: if both partA and partB are NaN, compare lexicographically
+
+    if (partA < partB) {
+      return -1;
+    } else if (partA > partB) {
+      return 1;
+    }
+  }
+
+  // If all parts are equal, compare the lengths of the section numbers
+  return partsA.length - partsB.length;
```

Comment:
Can there ever be a result different from 0?

Reply:
Assume files `3 foo.md` and `3.2 bar.md`, in this case the lengths of `partA = ["3"]` and `partB = ["3", "1"]` differ.

Reply:
Should I add a unit test for function `compareSectionNumbers`?